.github/actions/block-merging/action.yml

```
@@ -19,34 +27,84 @@ runs:
       env:
         BRANCH: ${{ inputs.branch }}
         BYPASS_PREFIX: ${{ inputs.bypass_prefix }}
+        FILTER: ${{ inputs.filter }}
+        MAX_WAIT_TIME: ${{ inputs.max_wait_time }}
         PR_TITLE: ${{ github.event.pull_request.title }}
       with:
         github-token: ${{ inputs.github_token }}
         script: |-
-          const {owner,repo} = context.repo;
+          const { owner, repo } = context.repo;
           const branch = process.env.BRANCH;
-
+          const filter = process.env.FILTER;
           const prTitle = process.env.PR_TITLE;
           const bypassPrefix = process.env.BYPASS_PREFIX;
 
-          if (prTitle.startsWith(bypassPrefix)) {
-            console.log(`Merging is allowed as the PR title ("${prTitle}") matches the bypass prefix ("${bypassPrefix}")`);
-            process.exit(0);
-          }
+          const filterRegex = new RegExp(filter);
 
-          const sleep = ms => new Promise(r => setTimeout(r, ms));
+          const maxWaitTimeString = process.env.MAX_WAIT_TIME;
+          const { hours, minutes, seconds } =
+            /^(?<hours>[0-9]*)h?(?<minutes>[0-9]*)m?(?<seconds>[0-9]*)s?$/.exec(
+              maxWaitTimeString,
+            ).groups;
+          const maxWaitTimeMs = (hours || 0) * 3600000 +
+            (minutes || 0) * 60000 +
+            (seconds || 0) * 1000;
```

This is perhaps clearer:

```suggestion
          const maxWaitTimeMs = (((hours || 0) * 60 + (minutes || 0)) * 60 + (seconds || 0)) * 1000;
```

Just a detail of course.